src/main/java/com/example/server/domain/diary/application/DiaryService.java

```diff
@@ -0,0 +1,96 @@
+package com.example.server.domain.diary.application;
+
+import static com.example.server.global.exception.ErrorCode.*;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.multipart.MultipartFile;
+
+import com.example.server.domain.diary.api.dto.DiarySaveRequest;
+import com.example.server.domain.diary.model.Diary;
+import com.example.server.domain.diary.model.DiaryStamp;
+import com.example.server.domain.diary.model.constants.Category;
+import com.example.server.domain.diary.model.constants.ChallengeTopic;
+import com.example.server.domain.diary.persistence.DiaryRepository;
+import com.example.server.domain.image.model.Image;
+import com.example.server.domain.member.application.MemberFindService;
+import com.example.server.domain.member.model.Member;
+import com.example.server.domain.pet.application.PetFindService;
+import com.example.server.domain.pet.model.Pet;
+import com.example.server.domain.stamp.application.StampFindService;
+import com.example.server.domain.stamp.model.Stamp;
+import com.example.server.domain.stamp.model.constants.StampType;
+import com.example.server.global.exception.BusinessException;
+import com.example.server.global.util.S3Uploader;
+
+import lombok.RequiredArgsConstructor;
+
+@Service
+@RequiredArgsConstructor
+public class DiaryService {
+
+	private final MemberFindService memberFindService;
+	private final PetFindService petFindService;
+	private final DiaryRepository diaryRepository;
+	private final StampFindService stampFindService;
+	private final S3Uploader s3Uploader;
+	static final int IMAGE_LIST_SIZE = 5;
+
+	@Transactional
+	public void createDiary(String username,
+		DiarySaveRequest diarySaveRequest,
+		List<MultipartFile> multipartFiles,
+		String dirName
+	) throws IOException {
```

Comment:
이렇게 다수의 파라미터가 들어올 때, 파라미터별로 개행을 주어 자르는 방식을 저도 매우 선호합니다!

다만, 첫 번째 파라미터부터 괄호 밑으로 잘라서 설계한다면, 더욱 파라미터가 직관적으로 인식될 것 같아요.

```suggestion
    public void createDiary(
            String username,
            DiarySaveRequest diarySaveRequest,
            List<MultipartFile> multipartFiles,
            String dirName
    ) throws IOException {
```